### PR TITLE
Allow user to change Globus origin.

### LINKED ISSUE #3196: Allow user to change Globus file location option so that none of Oak, Sherlock, or GDrive are selected
If a depositor initially selects Oak, Sherlock, or Google Drive as the location for their content to transfer and then later changes their mind and wants to transfer from a personal computer or another server on campus, they need to be able to do that and the current UI does not allow that.

Steps to recreate:
1. create new item
2. choose the Globus option
3. select Oak, Sherlock, or Google Drive from the drop-down menu
4. Save as draft
5. Click pencil icon to edit
6. under the Globus option, the user now only has the ability to select Oak, Sherlock or GoogleDrive and not the "Select..." option to blank this out. They no longer can choose to upload files to Globus via their personal computer or some other campus server.
7. choose a different file upload option, then select Globus option again -- still can't unchoose an option from the dropdown
8. choose a different file upload option
9. save as draft
10. edit item
11. choose Globus upload option -- again, still can't unchoose the options from the dropdown so they can upload from a personal computer or another server.

User needs to be able to select an option that doesn't use Oak, Sherlock, or Google Drive for the Globus upload.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3196

# Why was this change made? 🤔

# How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

Locally

https://github.com/sul-dlss/happy-heron/assets/588335/5aead8b5-88d9-4cc2-8eb9-02478969bd81

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

**Review by mjgiarlo — APPROVED** [review]

File: app/components/works/add_files_component.html.erb [hunk 1]
```diff
@@ -10,9 +10,9 @@
   <div class="form-check" data-complex-radio-target="selection">
     <%= form.radio_button :upload_type, "browser", class: "form-check-input", "data-file-uploads-target": "browserRadioButton",
           data: {action: "complex-radio#disableUnselectedInputs file-uploads#updatePanelVisibility deposit-button#updateDepositButtonStatus"} %>
-    <%= form.label :upload_type_browser, "Upload fewer than 250 files (total less than 10 GB) to be displayed as a flat list of files.", class: "form-check-label fw-semibold" %>
+    <%= form.label :upload_type_browser, "Upload one or more files (less than 10GB) to be displayed as a flat list of files.", class: "form-check-label fw-semibold" %>
```
Comment (jmartin-sul) [inline app/components/works/add_files_component.html.erb:13]: was catching up on PRs i missed from earlier today, and i'm wondering if this was an accidental reversion of [this change](https://github.com/sul-dlss/happy-heron/pull/3199/files#diff-8891d5301b70a836983b39503029abbd3839576f77e9280c10d413a92a9b48aaR13), possibly when rebasing?  (also totally possibly i'm confused or missing something)

File: app/components/works/add_files_component.html.erb [hunk 2]
```diff
@@ -10,9 +10,9 @@
...
     <div id="uploaded-files-panel" data-file-uploads-target="fileUploads">
-      <div data-controller="dropzone" data-dropzone-max-file-size="10000" data-dropzone-max-files="250" data-dropzone-previews-container=".dropzone-files-previews">
+      <div data-controller="dropzone" data-dropzone-max-file-size="10000" data-dropzone-previews-container=".dropzone-files-previews">
```
Comment (jmartin-sul) [inline app/components/works/add_files_component.html.erb:15]: similar question here for [this change](https://github.com/sul-dlss/happy-heron/pull/3199/files#diff-8891d5301b70a836983b39503029abbd3839576f77e9280c10d413a92a9b48aaR15)

**Review by jmartin-sul — COMMENTED** [review]
late question about changes that seem to partially undo #3199 (cc @justinlittman @mjgiarlo)

File: app/components/works/add_files_component.html.erb (hunk 1 above)
Comment (justinlittman) [reply app/components/works/add_files_component.html.erb:13]: Thanks for the catch. Reverted in https://github.com/sul-dlss/happy-heron/pull/3238

Comment (jmartin-sul) [reply app/components/works/add_files_component.html.erb:13]: 🙌

**Pull request merged**